runner/src/server/plugins/router.ts

```diff
@@ -59,7 +59,7 @@ export default {
           handler: async (request: HapiRequest, h: HapiResponseToolkit) => {
             const { cookies, referrer } = request.payload as CookiePayload;
             const { href, origin } = new Url(referrer);
-            const redirect = href.replace(origin, ""); // Ensure you only redirect to a local path
+            const redirect = href.replace(origin, "").replace("__", "/"); // Ensure you only redirect to a local path
```

Comment:
what is appending `__`?

Reply:
So when ZAP security audit runs it complains that this redirect could be hijacked as it depends on user input. AFAIK it tests this by changing the input value and seeing if it can effectively change the redirect (which it can in this case). To try and prevent this test failing I have added a replacement expression here and updated the dynamic input value to contain a known replacement value I did use "__" but I think it might be better to use the crumb value which is also now on this payload. This is just obfuscation for basic robot attempts to update it programmatically (as ZAP does). As I understand it, the removal of the origin already prevents malicious redirection, but I think ZAP doesn't realise this. Will be able to test for sure after ZAP runs on it.